doc/filters.md

```diff
@@ -0,0 +1,188 @@
+# CKAN Query Filters spec
+
+The goal of this specification is to provide a common syntax for defining filters
+that need to be applied to queries performed. It is not tied to a particular API
+endpoint or search backend, and it aims to be simple and extensible.
+
+## Overview
+
+Filters are always provided as a dictionary or as a list of dictionaries.
+
+When provided as a dictionary, the top level keys can either be:
+
+* A field name
+* A top level operator: one of `$or` or `$and`
+
+For example:
+
+```
+"filters": {
+    "field1": "value",
+    "$or": [
+        {"field2": ["value1", "value2"]},
+        {"field3": {"gte": 9000, "lt": 1000}}
+    ]
+}
```

Comment:
If we can use examples like this it could make the docs more engaging:

```suggestion
"filters": {
    "incident": "noise complaint",
    "$or": [
        {"resolution": ["unresolved", "in progress"]},
        {"sector": [2, 8, 10, {"gte":15, "lte": 30}, 35]}
    ]
}
```